torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6017,103 +5974,50 @@ def sample_inputs_i0_i1(op_info, device, dtype, requires_grad, **kwargs):
         # NOTE: `i0e`'s first-order gradient is not continous
         # at `0`, hence we don't test `i0e` with any input being `0`.
         # TODO: Remove this when `make_tensor` supports excluding `0`.
-        with torch.no_grad():
-            for sample in samples:
-                t = sample.input
-                t[t == 0] = torch.finfo(dtype).eps  # type: ignore[index]
+        for sample in samples:
+            t = sample.input
+            t[t == 0] = torch.finfo(dtype).eps  # type: ignore[index]
     elif requires_grad and op_info.op != torch.special.i0e:
         # Special Case for gradient
         # Sample with `0` in the input
         t = make_tensor((S,), device, dtype,
                         requires_grad=requires_grad)
-
-        with torch.no_grad():
-            t[0] = 0
+        t[0] = 0
 
         samples += (SampleInput(t),)  # type: ignore[assignment]
 
     return samples
 
 
-def sample_inputs_rsub(op_info, device, dtype, requires_grad, variant='tensor', **kwargs):
-    def _make_tensor_helper(shape, low=None, high=None):
-        return make_tensor(shape, device, dtype, low=low, high=high, requires_grad=requires_grad)
-
-    def _samples_with_alpha_helper(args, alphas, filter_fn=lambda arg_alpha: True):
-        filtered_product = filter(filter_fn, product(args, alphas))  # type: ignore[var-annotated]
-        return (SampleInput(input.detach().clone().requires_grad_(requires_grad),
-                            args=(arg,), kwargs=dict(alpha=alpha))
-                for (input, arg), alpha in filtered_product)
+def sample_inputs_rsub(op_info, device, dtype, requires_grad, other_scalar, **kwargs):
```

Comment:
Don't we have generic elementwise binary sample inputs now for ops like add and sub? Can this just reuse that function?

Reply:
We do have `sample_inputs_rbinops`. Now, it seems rather simple. I wonder whether we should use a more involved scheme like hte one proposed here that tests for tensor-tensor / scalar-tensor operations with different dtypes and all. What do you think?